src/k8s-engine/lima.ts

```diff
@@ -811,15 +838,28 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     });
   }
 
-  async start(config: { version: string; memoryInGB: number; numberCPUs: number; port: number; }): Promise<void> {
-    this.cfg = config;
+  // TODO: Put this in k3sHelper
+  createImageEventHandler(engineName: ContainerEngine) {
+    const imageProcessor = createImageProcessorFromEngineName(engineName, this);
+
+    if (!imageProcessor) {
+      throw new Error(`createImageEventHandler: No image processor for ${ engineName }`);
+    }
+    this.#imageEventHandler = new ImageEventHandler(imageProcessor as ImageProcessor);
+  }
+
+  async start(fullConfig: Settings): Promise<void> {
+    const config = this.cfg = fullConfig['kubernetes'];
     const desiredShortVersion = await this.desiredVersion;
     const previousVersion = (await this.currentConfig)?.k3s?.version;
     const isDowngrade = previousVersion ? semver.gt(previousVersion, desiredShortVersion) : false;
 
     this.#desiredPort = config.port;
     this.setState(K8s.State.STARTING);
     this.currentAction = Action.STARTING;
+    this.changedContainerEngine = this.#currentContainerEngine !== config.containerEngine;
+    this.#currentContainerEngine = config.containerEngine;
+    this.setupImageProcessor(fullConfig.images.namespace);
```

Comment:
No, this is as desired. The object model for a K8sBackend is:
K8sBackend -> 1 ImageEventHandler (set in background in `startBackend`) -> 1 imageProcessor

The backend can also access the current `imageProcessor` via `this.#imageProcessor`.

At first I set the namespace on the image after calling `setupImageProcessor` but TS was complaining about `this.imageProcessor` possibly being null (it can't be at that point), so I passed it in.